packages/edit-site/src/components/editor/index.js

```diff
@@ -192,31 +191,16 @@ export default function Editor( { listViewToggleElement, isLoading } ) {
 		! isLoading &&
 		( ( postWithTemplate && !! contextPost && !! editedPost ) ||
 			( ! postWithTemplate && !! editedPost ) );
-	const mode = useMemo( () => {
-		if ( isViewMode ) {
-			return postWithTemplate ? 'template-locked' : 'all';
-		}
-
-		if ( isEditMode && pageContentFocusType === 'hideTemplate' ) {
-			return 'post-only';
-		}
 
-		if ( postWithTemplate && hasPageContentFocus ) {
-			return 'template-locked';
+	// This is the only reliable way I've found to reinitialize the rendering mode
```

Comment:
Can you explain a bit more the issue?

Reply:
There's no issue, we want the exact behavior of this useEffect:

 - When moving from edit to view or view to edit (canvas mode), we want to reset the value of the rendering mode. But the value depends on the entity we're editing. If it's a page it should be "template-locked", otherwise it should be "all".

I don't like using useEffect for this kind of data logic because it's dependent on component rendering and it's not very reliable. Ideally these things are bundled in the data stores and things like that but with the constaints we have here (editor being reusable between edit-post and edit-site...) I couldn't find a better approach.

Reply:
Thanks for explaining!